arcgis-ios-sdk-samples/Cloud & Portal/Integrated Windows Authentication/IntegratedWindowsAuthenticationPortalMapBrowserViewController.swift

```diff
@@ -0,0 +1,200 @@
+//
+// Copyright © 2019 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import UIKit
+import ArcGIS
+
+/// A view controller for browsing maps in a portal.
+class IntegratedWindowsAuthenticationPortalMapBrowserViewController: UITableViewController {
+    /// The portal whose maps are displayed in the table view.
+    let portal: AGSPortal
+    
+    /// Creates a portal map browser with the given portal.
+    ///
+    /// - Parameter portal: A portal.
+    init(portal: AGSPortal) {
+        self.portal = portal
+        super.init(style: .grouped)
+        portal.load { [weak self] (error) in
+            guard let self = self else { return }
+            if let error = error {
+                self.portalDidFailToLoad(with: error)
+            } else {
+                self.portalDidLoad()
+            }
+        }
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    /// The portal items to display in the table view.
+    private var mapPortalItems = [AGSPortalItem]() {
+        didSet {
+            guard isViewLoaded else { return }
+            tableView.reloadData()
+        }
+    }
+    
+    /// Called in response to the portal loading successfully.
+    ///
+    /// - Parameter portal: The portal that succeeded to load.
+    func portalDidLoad() {
+        let parameters = AGSPortalQueryParameters(forItemsOf: .webMap, inGroup: nil)
```

Comment:
That's a good idea, but that wasn't in the sample design and none of the other platforms do that. I'll ask the team about it.

Reply:
It doesn't seem like the ability to paginate through additional results is an overly complex part of the API that would need showcasing in this context.
@njarecha are you thinking that by adding this, it would serve to highlight that the `limit` property merely exists?

Reply:
It is not for whether paginate is is overly complex, but if someone queries and don't see the web map they are expecting to see in the result then they will be puzzled. If there is an option to paginate then they will know right away that there are more web maps to query.

Reply:
@njarecha Apparently we have an open issue to demonstrate pagination functionality. Hopefully we'll implement that behavior in this or another sample in the not too distant future.